design-proposals/emts-scale-provisioning.md

```diff
@@ -0,0 +1,142 @@
+
+### MVP requirements
+
+Following are the MVP requirements for the scale provisioning of EMT-S edge node supporting OXM workflow:
+
+- Provision multiple BareMetal edge nodes without onboarding for the purpose of standalone/singleton use.
+- Provide deploy a service on the local network that can achieve this provisioning at scale.
+- Deploy the provisioning service on the local network that support PXE Boot (BIOS/UEFI with DHCP + TFTP) boot and iPXE with HTTPs.  
+- Have a UX to pre-register BareMetal edge nodes using Serial number or UUID or MAC address.
+- Provision different OS profiles to different edge nodes selected based on Serial number or UUID or MAC address.
+- Provision default OS when a device on the LAN boots over PXE and is not pre-registered.
+- Have a ux of  collecting provisioning logs and status of edge nodes.
+
+> Note: It might be possible for EMF-EIM to support provisioning of the EMT-S nodes. supporting this capability as part of MVP depends on any active customer requirements.
+
+## Solution
+
+The solution assumes that we will deploy a slimmed down version of EIM (aka EIM standalone) on a customer's premises (OXM warehouse). The slimmed-down EIM
+should only consist of required components to drive OS provisioning.
+
+### Slimmed-down EIM
+
+The provisioning of EMT-S at scale will be driven by a local orchestrator instance that will be slimmed down to include only necessary components. In a nutshell,
+it will consist of:
+
+- Foundational Platform Services that are required to deploy and run the orchestrator instance.
+- Limited observability stack that should only provide logs.
+- A reduced flavor of EIM - only infra-core and infra-onboarding will be deployed (no infra-managers as they won't be used by EMT-S).
+- Limited Web UI with EIM only.
+- No cluster and application orchestrator deployed.
+- Abandon HA requirements - EIM-S is primarily a single-node, on-prem deployment. Any data loss can be backed by hardware-level redundancy. Think of EIM-S as Rancher Desktop or Virtualbox.
+
+FPS components that are essential to EIM are:
+- IAM, Multi-Tenancy components
+- Keycloak
+- RS-proxy
+- Vault
+- secrets-config
+- MetalLB services
+
+The following FPS/Observability components should be disabled:
+- Kyverno
+- Prometheus and all metrics-related components (including infra-core's exporter, Mimir)
+- SRE exporter
+- Alerting monitors
+- Loki should be scaled down to minimal deployment
+
+**The installation of EIM-S should become a one-line command operation.**
```

Comment:
What are the assumption? A VM? A baremetal machine? a k8s cluster?

Reply:
let's go with the assumption we use the orch-installe for now and avoid going too deep into this optimization

Reply:
Resolving the comment, we will use new on-prem installer being designed in https://github.com/open-edge-platform/edge-manageability-framework/pull/290.